test/metrics/metrics_test.go

```diff
@@ -19,6 +19,72 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// TestMetricsWhenUsersAutomaticallyApproved verifies that `UserSignupsApprovedMetric` and `UserSignupsApprovedWithMethodMetric` counters are increased when users are approved
+func TestMetricsWhenUsersAutomaticallyApproved(t *testing.T) {
```

Comment:
Shouldn't we also rename `TestMetricsWhenUsersDeactivated()` to `TestMetricsWhenUsersManuallyApproved()`? Those two functions look identical and test both activation and deactivation but the names are a bit confusing...

Reply:
Ok, I renamed them to `TestMetricsWhenUsersManuallyApprovedAndThenDeactivated` and `TestMetricsWhenUsersAutomaticallyApprovedAndThenDeactivated`